### PR TITLE
Adding ability to assign if a user wants to close an issue.

### LINKED ISSUE #3154: Support assigning and closing in one comment
It'd be nice to have

    /assign
    /close

That way I can work in one comment instead of adding noise with two. Thanks!

### PR DESCRIPTION
**Pull request by brahmaroutu**
Fixes #3154

### CODE REVIEW

File: prow/plugins/close/close.go [hunk 1]
```diff
@@ -61,9 +64,20 @@ func handle(gc githubClient, log *logrus.Entry, ic github.IssueCommentEvent) err
...
+		if err := gc.AssignIssue(org, repo, number, []string{commentAuthor}); err != nil {
+			msg := "Assigning you to the issue failed."
+			if ok, merr := gc.IsMember(org, commentAuthor); merr == nil && !ok {
+				msg = "Only kubernetes org members may be assigned issues."
+			} else if merr != nil {
+				log.WithError(merr).Errorf("Failed IsMember(%s, %s)", org, commentAuthor)
+			} else {
+				log.WithError(err).Errorf("Failed AssignIssue(%s, %s, %d, %s)", org, repo, number, commentAuthor)
+			}
+			resp := fmt.Sprintf("you can't close an issue unless you authored it or you are assigned to it, error: %s", msg)
```
Comment (spxtr) [inline prow/plugins/close/close.go]: Don't include the error text itself.

File: prow/plugins/close/close_test.go [hunk 2]
```diff
@@ -110,14 +125,32 @@ func TestCloseComment(t *testing.T) {
...
 			shouldComment: true,
 		},
+		{
+			name:          "close by other person, assign and close",
+			action:        "created",
+			state:         "open",
+			body:          "/close",
+			commenter:     "non-owner",
+			shouldClose:   true,
+			shouldComment: false,
```
Comment (spxtr) [inline prow/plugins/close/close_test.go]: Add `shouldAssign`.

**Comment by spxtr** [comment]
Two minor comments, looks good otherwise.

**Comment by spxtr** [comment]
Two minor comments, looks good otherwise.

**Comment by spxtr** [comment]
/lgtm

Nice!

**Comment by cblecker** [comment]
@brahmaroutu @spxtr I'm not sure this is working: https://github.com/kubernetes/kubernetes/issues/50659#issuecomment-322536528

**Comment by spxtr** [comment]
I think we got behind on deploying prow while I was on vacation. I've redeployed hook so try again and let me know if it still doesn't work.

**Pull request merged**